prov/tcp/src/tcpx_shared_ctx.c

```diff
@@ -96,7 +96,8 @@ static ssize_t tcpx_srx_recvmsg(struct fid_ep *ep, const struct fi_msg *msg,
 		goto unlock;
 	}
 
-	tcpx_srx_recv_init(recv_entry, flags, msg->context);
+	tcpx_srx_recv_init(recv_entry, (srx_ctx->op_flags & FI_COMPLETION) | flags,
```

Comment:
The send/recv msg calls override the base op_flags associated with an EP. I think the original code is correct here.

Reply:
That sounds right.

Reply:
Not sure if this is relevant, but tcp might have to be updated with the util_ep->rx/tx_msg_flags that I added to OR into msg calls to signal the correct completion if we're using SELECTIVE_COMPLETION or op_flags